flake.nix

```diff
@@ -0,0 +1,86 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
```

Comment:
any reason to prefer unstable over 23.11 ?

Reply:
I have seen is most common in flake examples, I am not sure, but I think flakes offer a deterministic env via the `flake.lock` so that using unstable is not risky